app/Http/Controllers/Api/PirepController.php

```diff
@@ -116,8 +116,18 @@ protected function getFields(Request $request): ?array
             return [];
         }
 
+        $request_fields = $request->input('fields');
+
         $pirep_fields = [];
-        foreach ($request->input('fields') as $field_name => $field_value) {
+        $fillable = (new PirepFieldValue())->getFillable();
+        if (! empty(array_diff($fillable, $request_fields))) {
+            if (array_key_exists('fields', $request_fields)) {
+                $request_fields = $request->input('fields')['fields'];
+            } else {
+                return [];
+            }
+        }
+        foreach ($request_fields as $field_name => $field_value) {
```

Comment:
I'm not sure what you're doing here, the model already enforces fillable fields?

Reply:
Yes, but I have added validation as per the given object in your issue:

```
"fields": {
		"fields": {
			"key": "value"
		}
	}